aiogram/dispatcher/handler/base.py

```diff
@@ -23,18 +31,20 @@ def __init__(self, event: T, **kwargs: Any) -> None:
         self.data: Dict[str, Any] = kwargs
 
     @property
-    def bot(self) -> Bot:
+    def bot(self) -> Optional[Bot]:
         if "bot" in self.data:
-            return self.data["bot"]
+            # TODO: remove cast
```

Comment:
Can we add typing for BaseHandler.data?

Reply:
data has type `Dict[str, Any]` and more specific type can not be placed because anybody can add values to this mapping from filters and middlewares.

Reply:
So, what's better: use casting or isinstance?

Reply:
If needed to do different actions on different types - use isisntance(like raising exceptions or converting data and etc.) in other cases need to use casting